service/frontend/workflow_handler.go

```diff
@@ -4123,13 +4123,14 @@ func (wh *WorkflowHandler) getRawHistory(
 	var rawHistory []*commonpb.DataBlob
 	shardID := common.WorkflowIDToHistoryShard(namespaceID.String(), execution.GetWorkflowId(), wh.config.NumHistoryShards)
 
-	resp, err := wh.persistenceExecutionManager.ReadRawHistoryBranch(ctx, &persistence.ReadHistoryBranchRequest{
+	resp, err := wh.historyClient.ReadRawHistoryBranch(ctx, &historyservice.ReadRawHistoryBranchRequest{
```

Comment:
For this and the other APIs, can we use a dynamic config flag for this (with the value being an integer percentage for a gradual rollout) and also record the difference in performance in metrics? This is a big change, and I think cutting over all of our services to use this approach immediately is too risky. I know that the actual code looks identical, but a difference in the code is not the only concern. For example, what happens if a frontend is deployed with this new code that relies on the history services providing these APIs before any history service comes up that can serve that API?

Reply:
This is perhaps the most critical catch! It's so important that I can't even just put this behind a simply dynamic config flag! I will discuss with you offline.